src/components/FiltersDrawer/index.tsx

```diff
@@ -0,0 +1,126 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import Badge, { Variant } from "@leafygreen-ui/badge";
+import { palette } from "@leafygreen-ui/palette";
+import { SideNav, SideNavGroup } from "@leafygreen-ui/side-nav";
+import { Body } from "@leafygreen-ui/typography";
+import Cookie from "js-cookie";
+import Icon from "components/Icon";
+import { HAS_OPENED_DRAWER } from "constants/cookies";
+import { QueryParams } from "constants/queryParams";
+import { size, zIndex } from "constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import Filter from "./Filter";
+
+const { green, gray } = palette;
+
+interface FiltersDrawerProps {
+  ["data-cy"]?: string;
+}
+
+const FiltersDrawer: React.FC<FiltersDrawerProps> = ({ "data-cy": dataCy }) => {
+  const [collapsed, setCollapsed] = useState(
+    Cookie.get(HAS_OPENED_DRAWER) === "true"
+  );
+
+  const [filters, setFilters] = useQueryParam<string[]>(
+    QueryParams.Filters,
+    []
+  );
+
+  const deleteFilter = (filterName: string) => {
+    const newFilters = filters.filter((f) => f !== filterName);
+    setFilters(newFilters);
+  };
+
+  const editFilter = (oldFilter: string, newFilter: string) => {
+    // Duplicate filters are not allowed.
+    if (filters.includes(newFilter)) {
+      return;
+    }
+    const newFilters = [...filters];
+    const idxToReplace = newFilters.indexOf(oldFilter);
+    newFilters[idxToReplace] = newFilter;
+    setFilters(newFilters);
+  };
+
+  return (
+    <StyledSideNav
+      aria-label="Filters Side Nav"
+      collapsed={collapsed}
+      data-cy={dataCy}
+      setCollapsed={(collapse: React.SetStateAction<boolean>) => {
+        setCollapsed(collapse);
+        Cookie.set(HAS_OPENED_DRAWER, "true", { expires: 365 });
```

Comment:
Instead of using this as a flag, why not open the drawer initially and then use this to preserve whatever state the user leaves the drawer in? Seems like that may lead to the most consistent experience.

Reply:
if it's okay I would like to leave it like this for now 🥺 It should be easy to switch later!